src/flashbar/collapsible.scss

```diff
@@ -0,0 +1,229 @@
+    );
+    grid-row-start: 1;
+    grid-row-end: calc(var(#{custom-props.$flashbarStackIndex}) + 2);
+
+    @include stacked-flash-z-index;
+  }
+
+  > .item:not(:first-of-type) {
+    align-self: end;
+    /* Give placeholder elements in the stacked state the height of a notification
+       with one single line of text.
+       This makes them look better during the collapse animation, in which they are already empty.
+     */
+    min-height: calc(
+      #{awsui.$font-body-m-line-height} + (
+          #{awsui.$space-scaled-xs} + #{awsui.$border-field-width} + #{awsui.$space-scaled-xxs}
+        ) * 2
+    );
+  }
+
+  &.visual-refresh {
+    > .item:not(:last-child) > .flash,
+    > .item.flash {
+      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    }
+  }
+
+  &:not(.visual-refresh) {
+    > .item:not(:last-child) > .flash,
+    > .item.flash {
+      box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.15);
+    }
+  }
+}
+
+.stack > .expanded.animation-running > .flash-list-item {
+  position: relative;
+  @include stacked-flash-z-index;
+}
+
+@mixin toggle-button-layout($margin-top-when-collapsed, $border-width) {
+  $line-height: awsui.$font-body-m-line-height;
+  $padding-vertical: styles.$control-padding-vertical;
+  $margin-top-when-expanded: -12px;
+
+  padding-top: $padding-vertical;
+  padding-bottom: $padding-vertical;
+
+  &:not(.expanded) {
+    margin-top: #{$margin-top-when-collapsed};
+    margin-bottom: calc(
+      #{-$margin-top-when-collapsed} - (#{$line-height} + 2 * (#{$padding-vertical} + #{$border-width}))
+    );
+  }
+
+  &.expanded {
+    margin-top: #{$margin-top-when-expanded};
+    margin-bottom: calc(
+      #{-$margin-top-when-expanded} - (#{$line-height} + 2 * (#{$padding-vertical} + #{$border-width}))
+    );
+  }
+}
+
+.stack > .toggle {
+  @include styles.text-wrapping;
+
+  background: awsui.$color-background-layout-toggle-default;
+  border-color: awsui.$color-background-layout-toggle-default;
+  border-radius: awsui.$border-radius-button;
+  border-style: solid;
+  box-shadow: awsui.$shadow-panel-toggle;
+  color: awsui.$color-text-layout-toggle;
+  cursor: pointer;
+  display: flex;
+  grid-column: 2;
+  grid-row: 2;
+  gap: awsui.$space-xs;
+  letter-spacing: awsui.$font-button-letter-spacing;
+  margin-left: auto;
+  margin-right: auto;
+  text-decoration: none;
+  z-index: 1;
+
+  > .status > .text,
+  > .status > .types-count,
+  > .button {
+    @include typography.default-text-style;
+    color: awsui.$color-text-layout-toggle;
+    cursor: pointer;
+    margin-top: 0;
+    margin-bottom: 0;
+  }
+
+  > .status {
+    > .text {
+      font-weight: awsui.$font-button-weight;
+      display: inline;
+      margin-right: awsui.$space-s;
+    }
+
+    > .types-count {
+      margin-left: awsui.$space-xxs;
+
+      > .types-count-visible {
+        > .type-count {
+          margin-right: awsui.$space-s;
+
+          > .count-number {
+            margin-left: awsui.$space-xxs;
```

Comment:
I'm seeing a lot of spacing tokens in the CSS. Is there a reason why we want this way instead using `<SpaceBetween>` and `<Box>`?

Reply:
I guess there is no big reason. I might have thought this would involve a tiny tiny bit Iess JavaScript at runtime. Would using such components be preferable?

Reply:
It's an interesting question. I think in the past we definitely preferred components, but personally I prefer CSS as well. So... I'm okay with keeping it like this.